notificationworkerlambda/src/main/scala/com/gu/notifications/worker/delivery/fcm/FcmClient.scala

```diff
@@ -125,17 +125,19 @@ class FcmClient (firebaseMessaging: FirebaseMessaging, firebaseApp: FirebaseApp,
 }
 
 object FcmClient {
-  def apply(config: FcmConfig): Try[FcmClient] = {
+  def apply(config: FcmConfig, firebaseAppName: Option[String]): Try[FcmClient] = {
     Try {
       val firebaseOptions: FirebaseOptions = FirebaseOptions.builder()
           .setCredentials(GoogleCredentials.fromStream(new ByteArrayInputStream(config.serviceAccountKey.getBytes)))
           .setHttpTransport(new OkGoogleHttpTransport)
           .setConnectTimeout(10000) // 10 seconds
           .build
-      FirebaseApp.initializeApp(firebaseOptions)
+      firebaseAppName match {
+        case None => FirebaseApp.initializeApp(firebaseOptions)
+        case Some(name) => FirebaseApp.initializeApp(firebaseOptions, name)
```

Comment:
I understand why we're doing this, to give each firebase client a unique name. Is this a restriction of the firebase sdk? I'm wondering why we don't need equivalent functionality for apns?

Reply:
Yes, it is a [requirement of FirebaseSDK](https://firebase.google.com/docs/reference/admin/java/reference/com/google/firebase/FirebaseApp).  

I guess it may be just a matter of implementation.  For apns, it appears that Apple provides a web service URLs instead of a SDK (I may be wrong) and each call to the web service is probably independent.  We use third party library `pushy` and it does not have this restriction.